tools/local-environment/src/nomad.ts

```diff
@@ -672,216 +719,140 @@ export class Nomad {
+
+    const newBridgeDeploy = this.getBridgeDeploy(newNetwork, newCoreDeploy);
+    await deployNewChainBridge(newBridgeDeploy, govBridgeDeploy);
+    this.cacheDeploy(newBridgeDeploy, govBridgeDeploy);
 
-    const actualWatchers = newCoreDeploy.config.watchers.map((w) =>
-      w.toLowerCase()
+    this.setDeployed(newNetwork.domain);
+
+    this.updateArtifacts(
+      [newCoreDeploy, govCoreDeploy],
+      [newBridgeDeploy, govBridgeDeploy]
     );
 
-    const actualBridgeRouterAddress =
-      newBridgeDeploy.contracts.bridgeRouter!.proxy.address.toLowerCase();
-
-    let lastError: typeof AssertionError | undefined = undefined;
-    const w = new Waiter(
-      async () => {
-        try {
-          for (const [oldCoreDeploy, oldBridgeDeploy] of zippedDeploys) {
-            const govRouterAddress =
-              await oldCoreDeploy.contracts.governance!.proxy.routers(
-                newDomain
-              );
-            expect("0x" + govRouterAddress.slice(26).toLowerCase()).to.equal(
-              actualGovRouterAddress,
-              `Wrong remote GovernanceRouter address at Domain ${oldCoreDeploy.chain.domain}`
-            );
-
-            for (const wAddress of actualWatchers) {
-              const permissionExists =
-                await oldCoreDeploy.contracts.xAppConnectionManager!.watcherPermission(
-                  wAddress,
-                  newDomain
-                );
-              expect(
-                permissionExists,
-                `No permission exists for watcher '${wAddress}' and domain: ${newDomain}`
-              );
-            }
+    await this.updateMultiProvider();
 
-            const actualReplicaAddress =
-              oldCoreDeploy.contracts.replicas[
-                newDomain
-              ].proxy.address.toLowerCase();
-            const replicaAddress =
-              await oldCoreDeploy.contracts.xAppConnectionManager!.domainToReplica(
-                newDomain
-              );
-            expect(replicaAddress.toLowerCase()).to.equal(
-              actualReplicaAddress,
-              `Wrong Replica address at Domain ${oldCoreDeploy.chain.domain}`
-            );
-
-            const bridgeRouterAddress =
-              await oldBridgeDeploy.contracts.bridgeRouter!.proxy.remotes(
-                newDomain
-              );
-            expect("0x" + bridgeRouterAddress.slice(26).toLowerCase()).to.equal(
-              actualBridgeRouterAddress,
-              `Wrong remote BridgeRouter address at Domain ${oldCoreDeploy.chain.domain}`
-            );
-          }
+    // from here SDK will be used to send governance actions
 
-          return true;
-        } catch (e: any) {
-          lastError = e;
-        }
-      },
-      500_000,
-      5_000
+    await enrollSpoke(this.multiprovider!, newNetwork.domain, [
+      this.getWatcherKey(newNetwork)!.toAddress(),
+    ]);
+
+    // till here
+
+    this.updateArtifacts(
```

Comment:
why do we update artifacts and multiprovider before and after? they shouldn't change based on enrolling the spoke, should they?

Reply:
check `enrollSpoke` -- does it deploy also? or just enroll?

Reply:
u are right they actually shouldnt